lib/grandpa/message_test.go

```diff
@@ -113,16 +130,65 @@ func TestFinalizationMessageToConsensusMessage(t *testing.T) {
 	gs, err := NewService(cfg)
 	require.NoError(t, err)
 
-	fm, err := gs.newFinalizationMessage(gs.head, 77)
-	require.NoError(t, err)
+	gs.justification[77] = []*Justification{
+		{
+			Vote: &Vote{
+				hash:   common.Hash{0xa, 0xb, 0xc, 0xd},
+				number: 999,
+			},
+			Signature:   [64]byte{1, 2, 3, 4},
+			AuthorityID: gs.publicKeyBytes(),
+		},
+	}
 
+	fm := gs.newFinalizationMessage(gs.head, 77)
 	cm, err := fm.ToConsensusMessage()
 	require.NoError(t, err)
 
 	expected := &ConsensusMessage{
 		ConsensusEngineID: types.GrandpaEngineID,
-		Data:              common.MustHexToBytes("0x014d000000000000007db9db5ed9967b80143100189ba69d9e4deab85ac3570e5df25686cabe32964a0000000000000000"),
+		Data:              common.MustHexToBytes("0x014d000000000000007db9db5ed9967b80143100189ba69d9e4deab85ac3570e5df25686cabe32964a0000000000000000040a0b0c0d00000000000000000000000000000000000000000000000000000000e7030000000000000102030400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000034602b88f60513f1c805d87ef52896934baf6a662bc37414dbdbf69356b1a691"),
 	}
 
 	require.Equal(t, expected, cm)
 }
+
+func TestJustificationEncoding(t *testing.T) {
+	just := &Justification{
+		Vote: &Vote{
+			hash:   common.Hash{0xa, 0xb, 0xc, 0xd},
+			number: 999,
+		},
+		Signature:   [64]byte{1, 2, 3, 4},
+		AuthorityID: [32]byte{5, 6, 7, 8},
```

Comment:
should we make these constants in the test file and add to the top, they are being used throughout...?